bundles/org.openhab.binding.evohome/src/main/java/org/openhab/binding/evohome/internal/api/ApiAccess.java

```diff
@@ -188,18 +192,18 @@ private <TOut> TOut doRequest(HttpMethod method, String url, Map<String, String>
      * @return The result of the request or null
      * @throws TimeoutException Thrown when a request times out
      */
-    private <TOut> TOut doAuthenticatedRequest(HttpMethod method, String url, Object requestContainer,
-            Class<TOut> outClass) throws TimeoutException {
-        Map<String, String> headers = null;
-        if (authenticationData != null) {
-            headers = new HashMap<>();
-
-            headers.put("Authorization", "Bearer " + authenticationData.getAccessToken());
-            headers.put("applicationId", applicationId);
+    private @Nullable <TOut> TOut doAuthenticatedRequest(HttpMethod method, String url,
+            @Nullable Object requestContainer, @Nullable Class<TOut> outClass) throws TimeoutException {
+        Map<String, String> headers = new HashMap<>();
+        Authentication localAuthenticationData = authenticationData;
+        String localApplicaitonId = applicationId;
+
+        if (localAuthenticationData != null && localApplicaitonId != null) {
```

Comment:
Behavior is changed, is it intentional? Previously `applicationId` header would be set to null, and other headers set correctly. Now no headers will be set, but request will still be made (without any headers).

Reply:
Think i posible wrong an assumption here. Changed it, thanks.